### PR TITLE
Functionalities for chat history creation and storage, along with FastAPI's SSE-based endpoint to fetch conversations corresponding to a unique session ID

### LINKED ISSUE #13: Implement chat history creation and storage functionalities, along with FastAPI's SSE-based endpoint to fetch conversations corresponding to a unique session ID

### PR DESCRIPTION
**Pull request by MalayJoshi-Karya**
This PR introduces:

1. Backend Implementation for Chat History Storage and Fetching Feature:
- Endpoint /fetch_history: This FastAPI-based endpoint streams chat history for a given session using Server-Sent Events (SSE).
The endpoint accepts a POST request with a session_id in the request body and streams the chat history as SSE.
Chat history is retrieved from a SQLite database, with real-time streaming enabled for interaction updates.

- Chat History Management:
A SQLite database is used to store and manage the chat history for each session.
Functions create_chat_history_table(), store_chat_history(), and get_chat_history() handle the creation of the chat history table, storing chat interactions, and fetching chat history respectively.

- Chat History Table:
The chat history is stored in a SQLite table with columns: session_id, query, response, and timestamp.
This allows efficient storage and retrieval of session-specific data, maintaining a structured format for streaming.

2. Input and Output Formats:
- Input Request Format for fetching chat history:
{
    "session_id": "<your-session-id>"
}

- Output Response Format for the SSE stream:
{
    "session_id": "<session-id>",
    "query": "<original-query>",
    "response": "<generated-response>",
    "timestamp": "<timestamp-of-interaction>"
}

3. Prototype Client-Side Script:
- A prototype client-side Python script that demonstrates how the frontend can interact with the backend using SSE to retrieve chat history and real-time updates.
- This script is intended as a visualization of the interaction flow but is not the final version for frontend integration.

This commit closes #13

### CODE REVIEW

**Review by Mirza-Aasim — APPROVED** [review]
LGTM!

**Review by SoumyaRanjanPatnaik — CHANGES_REQUESTED** [review]

File: backend/fastapi-sse/server.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+logger = setup_logging('server', 'server_success.log', 'server_error.log')
+
+
+# Create the FastAPI app
+app = FastAPI()
+    
+# Configure CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],  # Allows all origins
```
[the remark this answers is not in the document]
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/server.py]: Done

File: backend/fastapi-sse/server.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    The request body should contain:
+    - query: The natural language query to process.
+    - session_id: Optional session identifier; if not provided, one will be generated.
+    - type: Optional type of query response (default is "report").
+    
+    Returns:
+        StreamingResponse: The response streamed as Server-Sent Events.
+    """
+    try:
+        body = await request.json()
```
[the remark this answers is not in the document]
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/server.py]: Done

File: backend/fastapi-sse/server.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    The request body should contain:
+        - session_id: The session identifier to fetch history.
+   
+    Returns:
+        StreamingResponse: The chat history streamed as Server-Sent Events (SSE).
+    """
+    try:
+        # Extract the session_id from the request body
+        body = await request.json()
```
[the remark this answers is not in the document]
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/server.py]: Done

File: backend/fastapi-sse/chat_history.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    cursor = conn.cursor()
+    cursor.execute('''
+    INSERT INTO chat_history (session_id, query, response)
+    VALUES (?, ?, ?)
+    ''', (session_id, query, response))
+    conn.commit()
+    conn.close()
+
+
+def get_chat_history(session_id: str) -> Generator[str, None, None]:
```
[the remark this answers is not in the document]
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/chat_history.py]: Done

**Review by SoumyaRanjanPatnaik — APPROVED** [review]
LGTM

**Pull request merged**